app/services/versioned_files_service/withdraw_action.rb

```diff
@@ -0,0 +1,33 @@
+class VersionedFilesService
+  # Deletes head version.
+  class WithdrawAction
+    # @param version [String] the version number
+    # @param object [VersionedFilesService::Object] the object
+    def initialize(version:, object:)
+      @version = version
+      @object = object
+    end
+
+    # @raise [UnknowVersionError] if the version is not found
+    # @raise [Error] if the version is not the head version
+    def call
+      raise VersionedFilesService::UnknowVersionError, "Version #{version} not found" unless version?(version:)
+      raise VersionedFilesService::Error, "The head version cannot be withdrawn, only be deleted" if head_version == version
+
+      # Update the version manifest.
+      version_manifest.withdraw(version:)
+      # Delete the cocina version file and set the new head symlink.
```

Comment:
Cocina, public xml, and content files shouldn't be changed. (Just the version manifest.)